### DISCUSSION TITLE
Automatic RTMP Republishing

### DISCUSSION THREAD

**Discussion by djaouedkh** [OvenMediaLabs/OvenMediaEngine#1754] [Help] [answered]
Hello, I’m new to OvenMediaEngine.

I send a test stream to OME using the following command:
ffmpeg -re -i ./video_test.mp4 -c copy -f flv -y rtmp://ome.stationdrone.net:1935/app/test 

OME successfully intercepts the stream and republishes it automatically in WebRTC and LLHLS, which are accessible via:
WebRTC: wss://OME_SERVER:3334/app/test
LLHLS: https://OME_SERVER:3334/app/test/llhls.m3u8

Problem:
I want OME to automatically republish the stream in RTMP as well, just like it does for WebRTC and LLHLS. The RTMP stream is needed for another backend server to process the stream.

However, when I try to access the RTMP stream, it does not work. Here are the commands I tried:
ffmpeg -i rtmp://OME_SERVER:1935/app/rtmp/test -c copy -f null -
or
ffplay rtmp://OME_SERVER:1935/app/rtmp/test
-----> Both commands fail to access the RTMP stream.

Configuration Files:
Server.xml
![image](https://github.com/user-attachments/assets/6cc6b1af-f8e2-4613-96a7-f1a81fab5a69)

stream_map.xml
![image](https://github.com/user-attachments/assets/8952bb52-e6d1-4d6e-a8de-7f1735601fd0)

OME version : 0.17.3

How can I configure OME to automatically republish the stream in RTMP, ensuring it is accessible similarly to WebRTC and LLHLS? Is there any specific configuration or additional step I need to follow ?

**Comment by nums** [comment]
You can push RTMP stream to another service : https://airensoft.gitbook.io/ovenmediaengine/push-publishing

**Comment by djaouedkh** [comment]
My another service is not supposed to directly read the RTMP stream. Instead, it needs to send the RTMP URL to an AI algorithm for processing.

The goal is to have a single RTMP URL (like rtmp://OME_SERVER:1935/app/test) that is republished by OME, just as it does for WebRTC and LLHLS

Is it not possible to access the RTMP stream using a simple URL similar to WebRTC/LLHLS, such as:
wss://......./app/test
https://....../app/test/llhls.m3u8
rtmp://OME_SERVER/app/test ?

**Comment by nums** [comment, marked as answer]
OME does not provide rtmp publishing.

**Comment by nums** [comment]
@djaouedkh  You can check the last commit, SRT publisher has just been committed, so in a near future you can do what you want to do with SRT ingest on ffmpeg

**Reply by djaouedkh** [reply]
ok thanks that's good to know
